### DISCUSSION TITLE
Raspberrymatic als VM unter Proxmox 8.1.x ARM auf OrangePI 5B

### DISCUSSION THREAD

**Discussion by steffenxx** [OpenCCU/OpenCCU#2857] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Ich habe aktuell eine lauffähige Installation auf einem RPI4. Darauf läuft ein Proxmox 8.1.7 in der ARM Version. Dort läuft ein Raspberrymatic in einer VM und ein Docker in einem LXC CT. Läuft eigentlich stabil. Ich wollte dem Ganzen eine wenig mehr Power gönnen und haben einen OrangPi 5B mit 16GB Ram und 256 emmc gekauft. Zur Stunde muss ich sagen, ich habe es ein wenig bereut, da sich das zu einem großen Bastelprojekt entwickelt hat. Ich bekomme Proxmox und Docker problemlos ans laufen. Aber ich scheitere an Raspberrymatic. Ich habe das System bereits mit dem Orangpi Debian Bookworm mit 6er Kernel aufgesetzt und mit einem aktuellen Standard Debian. In beiden Fällen rennt Raspberrymatic nach der Installation per Script und Neustart in 100% CPU Load und das wars dann. Ich habe es dann wie hier beschrieben, mit Armbian für opi5b und dem 5er Kernel probiert. Proxmox bekomme ich installiert, das Raspberrymatic Script startet hier aber einfach erst gar nicht. 

Mir ist klar das meine Angaben keine Präzise Fehleranalyse zulassen. 

Mich würden jetzt und hier aber erst mal ein paar Punkt weiterbringen?
1.) Läuft Raspberrymatic überhaupt auf einem OrangePi 5 mit Rockchip 3588 ?
2.) Welches OS ist das Richtige?
3.) Welchen Installationsweg sollte man wählen?
4.) Was muss alles für diese Hardware an Anpassungen gemacht werden.

    

### Describe the behavior you expected

s.o. 

### Steps to reproduce the issue

1.
2.
3.
...

### What is the version this bug report is based on?

3.75.7.20240420

### Which base platform are you running?

generic-aarch64 (Generic appliance, ARM64/aarch64)

### Which HomeMatic/homematicIP radio module are you using?

HmIP-RFUSB

### Anything in the logs that might be useful for us?

```txt
Wenn wir tiefer einsteigen müssen, kann ich Logs liefern.
```

**Comment by steffenxx** [comment]
Hab die Ursache gefunden. Scheinbar fehlt in den Debain Distributionen für den OrangePi 5(b) der "QEMU guest Agent". Dementsprechend war das Problem schnell mit einem

`apt install qemu-guest-agent`

und einem Reboot gelöst. Läuft jetzt 1A auf "Orangepi5b_1.0.8_debian_bookworm_server_linux6.1.43" :-)

**Comment by jens-maus** [comment]
Die Problem/Issuebeschreibung verstehe ich nicht wirklich und erst recht nicht den letzten Beitrag mit dem qemu-guest-agent? Inwiefern hat das jetzt was mit RaspberryMatic zu tun?

**Comment by steffenxx** [comment]
Ich hatte gehofft das könntest du mir sagen, so tief stecke ich da nicht drin. Ich habe nur beobachtet, das unter der o.g. Hard- und Software Konfiguration, die Raspberrymatic VM startet, auf der Konsole erscheint "Display not aktiv" (oder so), in der Übersicht steht bei IP "Gast Agent nicht aktiv" und die CPU Last geht auf 100%. 

Daran ändert sich nichts. Man kann die VM nicht mal runterfahren. Nur ein "Stop" beendet das Ganze. 

Nachdem ich das qemu-guest-agent Paket installiert hatte, läuft alles wie es soll. 

Unter Armbian, erscheint beim Start des Scripts nur die erste Zeile, es wechselt dann aber nicht in den Dialog. 

Das Ganze scheint irgendwie mit dem OrangePi bzw. mit den für den OPI gemachten Debian bzw. Armbian Versionen zusammen zu hängen. Wie ich geschrieben habe, lieft auf einem Raspi4 alles auf Anhieb wie beschrieben durch.

**Comment by jens-maus** [comment]
Es ist wirklich schwer deinen Auführungen zu folgen. Vielleicht sollten wir erst einmal rausfinden WAS genau du da versuchst zu tun bzw. wie du versuchst RaspberryMatic zu installieren. Auf der Einen Seite schreibst du von Proxmox, dann von Debian, dann von VM, dann von LXC-CT und dann noch von qemu-guest-agent paket usw. Das ergibt aber zusammengenommen so erst einmal wenig Sinn.

Ich nehme erst einmal an du hast ein Proxmox 8 unter einem OrangePi installiert, richtig? Unter welchen Grund-OS hast du das getan? Mit welchem Grund-OS also? Und ich schätze du bist anhand der entsprechenden "Promox (ARM)" Anleitung vorgegangen (siehe https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE#proxmox-ve-arm-grundinstallation). Richtig?

Und wenn das richtig ist, WIE genau versuchst du jetzt unter einem Proxmox (ARM) RaspberryMatic ans laufen zu bekommen? Als VM oder als LXC Container oder gar als Docker Container innerhalb einer weiteren Debian VM?

Also bitte erst einmal näher auf dein Einvironment eingehen und alles etwas systematischer/logischer erklären hier...

**Comment by steffenxx** [comment]
Hallo Jens, sorry wenn ich dich verwirre. Liegt evtl. daran, das ich seit 2 Wochen an dem Thema dran bin und so langsam am verzweifelt war. ;-) 

Was ich vorhabe ist eigentlich ganz simpel:

1.) Hardware: OrangePi 5b
2.) OS: Orangepi5b_1.0.8_debian_bookworm_server_linux6.1.43 -> Auf interne EMMC übertragen 
3.) Dann Proxmox 8 für ARM wie in dem Link beschrieben installiert
4.) vmbr0 angelegt, stand in einer anderen Proxmox 8 für ARM Anleitung 

5.) Dann möchte ich Raspberrymatic als VM unter Proxmox 8 mit einen HmIP-RFUSB Installieren. Dafür nutze ich wie beschrieben den Befehl: 

`wget -qO - https://raspberrymatic.de/install-proxmox.sh | bash -`

Die Installation läuft fehlerfrei durch. Wenn ich dann die VM starte, hängt sie wie oben beschrieben. 

Da du in deiner Seite eine Installation mit Armbian als Host OS (Schritt 1.) beschrieben hast, habe ich das auch mal probiert. Damit startet aber das sh Script (Schritt 5.) erst gar nicht. Ich hatte das Ganze auch mit einem Standard (nicht OrangePI) Debian versucht. Damit rannte ich aber in das gleiche Verhalten wie mit dem OrangePI Debian. 

Wie gesagt, eine Grundinstallation mit Rapberry PI OS Lite 64 Bit, Bookworm dann Proxmox 8 und dann dein VM Sh Script für Raspberrymatic läuft auf einem RPi4 problemlos durch und die Raspberrymatic VM startet dann auch sauber. Leider ist der 4er mit 4GB im vergleich mit dem OrangePI etwas schwach auf der Brust.  

Auch wenn es evtl. nicht logisch klingt, in meinem o.g. Fall mit OrangePI Debian als HOST OS, reichte es aus, einfach den qemu-agent zu installieren. Dabei werden ja eine ganze latte Pakete installiert, evtl. ist es ja eines von diesen die den Ausschlag gegeben haben. 

Aktuell läuft der OrangePI jedenfalls super mit einer Raspberrymatic Proxmox VM, einem Proxmox CT mit Docker in dem 15 Container laufen und einem Proxmox CT mit Tailscale für die Verbindung zu meinem Frontend Server im Internet. Auf dem auch ein Proxmox läuft.

**Comment by steffenxx** [comment, marked as answer]
issue kann geschlossen werden. Wir haben den Orange Pi verkauft und sind auf einen N100 Barebone umgestiegen. Leider gab es mit der Rockchip CPU ständig Probleme. Nichts hat "Out-of-the-Box" funktioniert. Wer also nicht nach jedem Update eine Nachtschicht einlegen will, lässt die Finger von einem Orange PI und nutzt eine "Standard" Hardware.
